src/main/jbake/content/blog/2017/07/jaeger-with-security-proxy.adoc

```diff
@@ -0,0 +1,253 @@
+only certain groups to access trace data, or you might have deployed Jaeger into a public cloud.
+In a true microservices way, one possible approach is to add a sidecar to the Jaeger Query service,
+acting as a security proxy. Incoming requests hit our sidecar instead of reaching Jaeger's Query service
+directly and the sidecar would be responsible for enforcing the authentication and authorization constraints.
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-07-18-teaser.png[Jaeger login screen]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-07-18-teaser.png[Jaeger login screen]
+endif::[]
+
+For demonstration purposes we'll make use of link:http://keycloak.org[Keycloak] as our security solution, but
+the idea can be adapted to work with any security proxy. This demo should also work without changes with
+link:https://access.redhat.com/products/red-hat-single-sign-on[Red Hat SSO]. For this exercise, we'll need:
+
+* A Keycloak (or Red Hat SSO) server instance running. We'll call its location `${REDHAT_SSO_URL}`
+* An OpenShift cluster, where we'll run Jaeger backend components. It might be as easy as `oc cluster up`
+* A local clone of the link:https://github.com/jaegertracing/jaeger-openshift/blob/master/production/jaeger-production-template.yml[Jaeger OpenShift Production template]
+
+Note that we are not trying to secure the communication between the components, like from the Agent to the Collector.
+For this scenario, there are other techniques that can be used, such as mutual authentication via certificates, employing
+link:https://istio.io/[istio] or other similar tools.
+
+=== Preparing Keycloak
+
+For this demo, we'll run Keycloak via Docker directly on the host machine. This is to stress that Keycloak
+does *not* need to be running on the same OpenShift cluster as our Jaeger backend. 
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-07-18-keycloak.png[Keycloak login screen]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-07-18-keycloak.png[Keycloak login screen]
+endif::[]
+
+The following command should start an appropriate Keycloak server locally. If you already have your
+own Keycloak or Red Hat SSO server, skip this step.
+
+```bash
+docker run --rm --name keycloak-server -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=password -p 8080:8080 jboss/keycloak
+```
+
+Once the Keycloak server is up and running, let's create a realm for Jaeger:
+
+* Login into Keycloak (http://<YOUR_IP>:8080/auth/admin/master/console) with `admin` as username and `password` as password
+* In the top left corner, mouse over the `Select realm` box and click `Add realm`. Name it `jaeger` and click `Create`
+* On `Clients`, click `Create` and set `proxy-jaeger` as the name and save it
+* Set the `Access Type` to `confidential` and `*` as `Valid Redirect URIs` and save it. You might want to fine tune this in a 
```

Comment:
On the previous step, the user clicks on "Save" and is directed to a screen where these fields should be available. Should I mentioned the screen name anyway, even if that would probably mean "the page you are currently on" on the usual case?